example/gmres/test_real_A.cpp

```diff
@@ -55,19 +55,23 @@
 
 int main(int /*argc*/, char ** /*argv[]*/) {
 
-  typedef double                             ST;
+  using ST = Kokkos::Experimental::half_t;
+  //using ST = double;
```

Comment:
Let's remove the half_t line?  Or make the half_t into a separate test?  I really want to leave this test in double precision.